src/main/kotlin/snyk/common/SnykCachedResults.kt

```diff
@@ -141,13 +141,19 @@ class SnykCachedResults(
                     currentContainerError = null
                 }
 
-                override fun scanningSnykCodeFinished(snykResults: Map<SnykFile, List<ScanIssue>>) {
+                override fun scanningSnykCodeFinished() {
+                }
+
+                private fun updateCodeCache(snykResults: Map<SnykFile, List<ScanIssue>>) {
                     currentSnykCodeResultsLS.clear()
                     currentSnykCodeResultsLS.putAll(snykResults)
                     logger.info("Snyk Code: scanning finished for project ${project.name}, assigning cache.")
                 }
 
-                override fun scanningOssFinished(snykResults: Map<SnykFile, List<ScanIssue>>) {
+                override fun scanningOssFinished() {
+                }
+
+                private fun updateOSSCache(snykResults: Map<SnykFile, List<ScanIssue>>) {
```

Comment:
This doesn't look correct - on receiving a diagnostic, you clear the whole cache - it should only clear the entry of the diagnostic received.